abixen-platform-modules/src/main/java/com/abixen/platform/module/chart/service/impl/DatabaseConnectionServiceImpl.java

```diff
@@ -110,20 +119,40 @@ public DatabaseConnection updateDatabaseConnection(DatabaseConnection databaseCo
 
     @Override
     public void testDatabaseConnection(DatabaseConnectionForm databaseConnectionForm) {
-        databaseService.getConnection(databaseConnectionForm);
+        chooseDatabaseService(databaseConnectionForm).getConnection(databaseConnectionForm);
     }
 
     @Override
     public List<String> getTables(Long databaseConnectionId) {
         DatabaseConnection databaseConnection = dataSourceConnectionRepository.findOne(databaseConnectionId);
+        DatabaseService databaseService = chooseDatabaseService(databaseConnection);
         Connection connection = databaseService.getConnection(databaseConnection);
         return databaseService.getTables(connection);
     }
 
     @Override
     public List<String> getTableColumns(Long databaseConnectionId, String table) {
         DatabaseConnection databaseConnection = dataSourceConnectionRepository.findOne(databaseConnectionId);
+        DatabaseService databaseService = chooseDatabaseService(databaseConnection);
         Connection connection = databaseService.getConnection(databaseConnection);
         return databaseService.getColumns(connection, table);
     }
+
+    private DatabaseService chooseDatabaseService(DatabaseConnectionForm databaseConnectionForm) {
+        return chooseDatabaseServiceByType(databaseConnectionForm.getDatabaseType());
+    }
+
+    private DatabaseService chooseDatabaseService(DatabaseConnection databaseConnection) {
+        return chooseDatabaseServiceByType(databaseConnection.getDatabaseType());
+    }
+
+    private DatabaseService chooseDatabaseServiceByType(DatabaseType databaseType) {
+        switch (databaseType) {
+            case POSTGRES:
+                return databasePostgresService;
+            case H2:
+                return databaseH2Service;
+            default: return null;
```

Comment:
Please throw DatabaseTypeException - make it similarly to DatabaseConnectionException.